zipline/data/data_portal.py

```diff
@@ -193,17 +196,21 @@ def get_current_price_data(self, asset, column):
             asset_file_index = daily_attrs['first_row'][str(asset_int)]
 
             # find when the asset started trading
-            asset_start_trading_date = \
-                self.asset_finder.retrieve_asset(asset).start_date
-
-            if self.current_day < asset_start_trading_date:
+            # TODO: only access this info once.
+            calendar = daily_attrs['calendar']
+            asset_data_start_date = \
+                pd.Timestamp(
+                    calendar[daily_attrs['calendar_offset'][str(asset_int)]],
+                    tz='UTC')
+
+            if dt < asset_data_start_date:
```

Comment:
got it.

I think you need to use it on line 221 as well - that's where we do the offset calc right?  otherwise the daily data returned is still using self.current_day